src/query/json.rs

```diff
@@ -77,20 +77,33 @@ impl FromQueryResult for JsonValue {
             crate::QueryResultRow::SqlxPostgres(row) => {
                 use serde_json::json;
                 use sqlx::{postgres::types::Oid, Column, Postgres, Row, Type};
+
                 for column in row.columns() {
                     let col = if !column.name().starts_with(pre) {
                         continue;
                     } else {
                         column.name().replacen(pre, "", 1)
                     };
                     let col_type = column.type_info();
+
                     macro_rules! match_postgres_type {
                         ( $type: ty ) => {
-                            if <$type as Type<Postgres>>::type_info().eq(col_type) {
-                                try_get_type!($type, col)
+                            match col_type.kind() {
+                                #[cfg(feature = "postgres-array")]
+                                sqlx::postgres::PgTypeKind::Array(_) => {
+                                    if <Vec<$type> as Type<Postgres>>::type_info().eq(col_type) {
```

Comment:
Match a vector of primitive against column type, since the column type is an array